### ISSUE TITLE
rpm-attr file permission for parent directory are ignored

### ISSUE THREAD

**Issue by LStuker** [jordansissel/fpm#1143]
If I specify rpm-attr for a parent directory it gets ignored in the spec file.

For example:

```bash
fpm -s dir -t rpm -n "myapp" -v 1.0 
--rpm-attr '0755,appuser,appgrp:"/var/lib/myapp/relation"
--rpm-attr '0755,appuser,appgrp:/var/lib/myapp/relation/tmp'
--rpm-attr '0755,appuser,appgrp:/var/lib/myapp/relation/data'
/opt /var/lib/myapp /etc/myapp/relation /usr/lib/systemd/system/
```

SPECS file output:
```
%files
%defattr(-,root,root,-)
%config(noreplace) /etc/myapp/relation/config.yml

%dir /opt/myapp/relation/
%dir /opt/myapp/relation/bin
# Reject config files already listed or parent directories, then prefix files
# with "/", then make sure paths with spaces are quoted. I hate rpm so much.
/opt/myapp/relation/bin/myappcmd
/usr/lib/systemd/system/myapp.service
%attr(0755,appuser,appgrp) /var/lib/myapp/relation/data

%attr(0755,appuser,appgrp) /var/lib/myapp/relation/tmp
```

At first I thought fpm doesn't support this because rpm doesn't support it. But I manually edit the SPEC files and the output rpm creates the directory permission as desire.

My code debugging showed that fpm, goes throw all files and check if they are in also definied in the attr hash, if so it would add it. But as subdirectories are never be part of the file list they will never be checked if they are defined in attrs.

https://github.com/jordansissel/fpm/blob/3338aa8f4b5ba90c1314f3c4887b33cee1c36f7e/templates/rpm.erb#L252
```ruby
<% files.each do |path| -%>
<%   path = "/#{path}" -%>
<%   next if config_files.include?(path)-%>
<%   next if directories.include?(path)-%>
<%= rpm_file_entry(path) %>
<% end -%>

[1] pry(#<FPM::Package::RPM>)> files
=> ["etc/myapp/relation/config.yml",
 "opt/myapp/relation/bin/myappcmd",
 "usr/lib/systemd/system/dhive-relation.service",
 "var/lib/myapp/relation/data",
 "var/lib/myapp/relation/tmp"]
```

https://github.com/jordansissel/fpm/blob/3338aa8f4b5ba90c1314f3c4887b33cee1c36f7e/lib/fpm/package/rpm.rb#L190

```ruby
def rpm_file_entry(file)

    original_file = file
    file = rpm_fix_name(file)

    if !attributes[:rpm_use_file_permissions?]

      if attrs[file].nil?
        return file
      else
        return sprintf("%%attr(%s) %s\n", attrs[file], file)
      end
    end

    return sprintf("%%attr(%s) %s\n", attrs[file], file) unless attrs[file].nil?

[1] pry(#<FPM::Package::RPM>)> attrs
=> {"/var/lib/myapp/relation"=>"0755,appuser,appgrp",
 "/var/lib/myapp/relation/tmp"=>"0755,appuser,appgrp",
 "/var/lib/myapp/relation/data"=>"0755,appusr,appgrp"}
```

Is there a reason why fpm behaves like this? Or could it be extend to support subdirectories?

**Comment by LStuker** [comment]
I am catching this up again. Because I saw no activity on this Pull request. Is there anything which I  help to process this pull request?

**Comment by dclendenan** [comment]
I'm seeing the same issue - rpm-attr on directories is being ignored.

**Comment by vinzent** [comment]
hit this one too.

**Comment by LStuker** [comment]
@jordansissel thank you very much for your effort to build fpm. It is awesome! 
Is it possible to give a short feedback to this issue and the linked pull request?

**Comment by eryabies** [comment]
Hi!

This issue hits me today when I want to create a custom Wildfly rpm from the Wildfly tar.gz.
`fpm -t rpm -a noarch -s dir -C wildfly -n pt_infra_wildfly -v ${VERSION} --iteration ${BUILD_NUMBER} --category "Applications/System" --url "http://www.wildfly.org" --license "LGPL 2.1" --description "WildFly is a flexible, lightweight, managed application runtime that helps you build amazing applications" --rpm-user wildfly --rpm-group wildfly --rpm-defattrfile 0640 --rpm-defattrdir 0750 opt/local/software/wildfly`

**Comment by carlcantprogram** [comment]
fpm version 1.9.2

confirmed broken for rpm packages with both:
--rpm-use-file-permissions
and
--rpm-group mygroup  --rpm-user myuser

Files owner and group are set as expected, directories owner and group are set to root

**Comment by zbindenren** [comment]
Same issue here. It does now work for directories.

**Comment by jordansissel** [comment]
This is good feedback! Thank you everyone for reporting details (please continue if you find cases where this doesn't work).

Hopefully I or someone else can get to resolving this soon.

**Comment by eryabies** [comment]
Hi!

Any updates or any suggestions how to change subdir owenership?

**Comment by cmcginty** [comment]
I don't think this is a bug. Make sure to add `--directories <DIR>` for any directory your package owns. Once you do that it will let you set `--rpm-attr` on the directory.

**Comment by rptaylor** [comment]
Oof it took some trial and error but I found that `--rpm-attr` does not apply attributes to a directory if there is a trailing slash on the directory.   It works this way for directories:  `--rpm-attr '0750,root,somegroup:/mydir'`
